src/Config.php

```diff
@@ -41,6 +41,8 @@ public function __construct($name = 'default')
 
     /**
      * @return static
+     *
+     * @deprecated since 3.0
```

Comment:
Since 2.17 :)

You should also add the following call inside the method:
```php
@trigger_error('some deprecation message', E_USER_DEPRECATED);
```
and add/update a unit test that proves the deprecation is triggered when the method is called.

Reply:
Oh, do we add deprecation tests? Didn't notice that ^^

Reply:
yes we do :)
for example ; https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.16.1/tests/Fixer/ArrayNotation/ArraySyntaxFixerTest.php#L40

Reply:
OK, updated as told. 

Hope I did everything right. ^^